bot/connector-teams/src/main/kotlin/TeamsConnector.kt

```diff
@@ -47,70 +50,76 @@ internal class TeamsConnector(
     private val connectorId: String,
     private val path: String,
     val appId: String,
-    appPassword: String
+    val appPassword: String
 ) : ConnectorBase(teamsConnectorType) {
 
     companion object {
         private val logger = KotlinLogging.logger {}
     }
 
-    private val client = TeamsClient(appId, appPassword)
+    private val client = TeamsClient()
     private val executor: Executor by injector.instance()
     private val authenticateBotConnectorService = AuthenticateBotConnectorService(appId)
 
     override fun register(controller: ConnectorController) {
+        launchTokenCollector(appId, appPassword)
```

Comment:
you should stop the crons in a Connector#unregister method

Reply:
Very good point